api/envoy/api/v2/cluster/circuit_breaker.proto

```diff
@@ -21,8 +22,23 @@ import "validate/validate.proto";
 message CircuitBreakers {
   // A Thresholds defines CircuitBreaker settings for a
   // :ref:`RoutingPriority<envoy_api_enum_core.RoutingPriority>`.
-  // [#next-free-field: 8]
+  // [#next-free-field: 9]
   message Thresholds {
+    message RetryBudget {
```

Comment:
Should we have runtime control here for tweaking percent/concurrency like we do for the other static circuit breakers? This seems like something we should have?

Reply:
Yes, but I wanted to do that in a subsequent patch to rework the resource manager to accomodate the retry budget rather than hacking it into the cluster. I can go ahead and add them with a TODO to move all of this into the resource manager.

Reply:
Is there any reason you can't just move it into the resource manager now? With the way the code is written now it seems like it wouldn't be that hard and would be a lot cleaner? WYDT?

/wait-any

Reply:
My initial concern was that we'd need to rethink the abstractions in a way that doesn't break the stats enabled by the `track_remaining` flag. I went ahead and took a stab at moving this into the resource manager, but imo it's still a bit awkward when it comes to the stats and budget abstraction. Maybe just take a look at what I did in the resource manager and let me know if that looks ok to you at a high level. If not, I can always just revert back to what was there before.

I think it does look cleaner now, though.